packages/core/src/lib/list-box.svelte

```diff
@@ -0,0 +1,249 @@
+    right: [...options.right, ...newRightOptions],
+  };
+
+  options = newOptions;
+};
+
+afterUpdate(addNewData);
+
+const handleOptionClick = (option: ListBoxOption, side: ListBoxSide) => {
+  if (disabled) {
+    return;
+  }
+
+  option.selected = !option.selected;
+  options = { ...options };
+
+  dispatch('option-click', {
+    target: { ...option, side },
+  });
+};
+
+const handleMoveClick = (target: ListBoxSide) => {
+  if (disabled) {
+    return;
+  }
+
+  const source = target === LEFT ? RIGHT : LEFT;
+
+  const newOptions: Options = {
+    left: [],
+    right: [],
+  };
+
+  for (const option of options[source]) {
+    if (option.selected) {
+      newOptions[target].push({ ...option, selected: false });
+    } else {
+      newOptions[source].push(option);
+    }
+  }
+
+  options[source] = newOptions[source];
+  options[target] = [...options[target], ...newOptions[target]];
+  options = { ...options };
+
+  // copy object to avoid any potential pass-by-reference issues
+  dispatch('move', {
+    options: JSON.parse(JSON.stringify(options)) as Options,
+  });
+};
+</script>
+
+<div
+  class={cx('grid w-full grid-cols-[1fr_32px_1fr] items-center gap-8 p-2', {
+    '!text-black/50': disabled,
+  })}
+>
+  <div
+    class="flex w-full flex-col gap-2 self-stretch"
+    style={`height: ${height};`}
+  >
+    <span class="text-sm text-subtle-1">{leftlabel}</span>
+    <div
+      class="flex grow flex-col overflow-auto border border-medium bg-white p-2"
+    >
+      {#if options.left.length > 0}
+        {#each options.left as option}
+          <button
+            class={cx('py flex items-center px-2 text-sm', {
+              'bg-focus/highlight': option.selected,
+            })}
+            on:click={() => handleOptionClick(option, LEFT)}
+          >
+            <input
+              type="checkbox"
+              checked={option.selected}
+              {disabled}
+            />
+            <span class="px-4">{option.value}</span>
+            {#if suffix && option.suffix}
+              <span class="text-subtle-2">{option.suffix}</span>
+            {/if}
+          </button>
+        {/each}
+      {:else}
+        <div class="text-sm">
+          <slot name="left-empty" />
+        </div>
+      {/if}
+    </div>
+  </div>
+  <div class="flex flex-col gap-4">
+    <button
+      class={cx(
+        'flex h-8 w-8 rotate-90 cursor-pointer items-center justify-center border border-black bg-white hover:scale-105',
+        { 'border-black/50': disabled }
+      )}
+      data-testid="move-right"
+      on:click={() => handleMoveClick(RIGHT)}
+    >
+      <Icon
+        class="relative bottom-0.5"
+        name="arrow-up"
+      />
+    </button>
+    <button
+      class={cx(
+        'flex h-8 w-8 -rotate-90 cursor-pointer items-center justify-center border border-black bg-white hover:scale-105',
+        { 'border-black/50': disabled }
+      )}
+      on:click={() => handleMoveClick(LEFT)}
+      data-testid="move-left"
```

Comment:
Why is this `testid` needed? Is there a way to avoid it with a different component structure or better queries?

Reply:
Ill remove and try and rework the tests

Reply:
Looks like both of the move buttons are almost identical. I tried some stuff to screen get them but was unsuccessful. If you have any ideas im down to try them.